operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java

```diff
@@ -71,15 +73,22 @@ public Controller(Reconciler<P> reconciler,
     eventSourceManager = new EventSourceManager<>(this);
 
     final var hasDeleterHolder = new boolean[] {false};
-    dependents = configuration.getDependentResources().stream()
-        .map(drs -> createAndConfigureFrom(drs, kubernetesClient))
-        .peek(d -> {
-          // check if any dependent implements Deleter to record that fact
-          if (!hasDeleterHolder[0] && d instanceof Deleter) {
-            hasDeleterHolder[0] = true;
-          }
-        })
-        .collect(Collectors.toList());
+    final var specs = configuration.getDependentResources();
+    final var size = specs.size();
+    if (size == 0) {
+      dependents = Collections.emptyMap();
+    } else {
+      final var dependentsHolder = new HashMap<String, DependentResource>(size);
+      specs.forEach((name, drs) -> {
+        final var dependent = createAndConfigureFrom(drs, kubernetesClient);
+        // check if dependent implements Deleter to record that fact
+        if (!hasDeleterHolder[0] && dependent instanceof Deleter) {
+          hasDeleterHolder[0] = true;
```

Comment:
I would populate a `List<Deleter>` instance in parrallel, this will avoid to stream/filter/cast in Cleanup.